ospfd/ospf_lsa.c

```diff
@@ -2220,22 +2220,6 @@ void ospf_external_lsa_refresh_default(struct ospf *ospf)
 	}
 }
 
-void ospf_default_originate_lsa_update(struct ospf *ospf)
```

Comment:
Possible to get a bit more discussion about why this is the incorrect fix in the commit message?  It's not clear to me why this is wrong.

Reply:
* the function is horribly misnamed considering what it does is remove LSAs from the LSDB.
* the LSDB code correctly handles the case of originating a new LSA when a MaxAge LSA is currently in the DB.  Removing the MaxAge LSA resets the sequence number, which means we can end up reusing the same sequence number, which means other routers may ignore the new LSA.
* also if we don't originate a new LSA, the MaxAge LSA may not be properly propagated to other routers (if it has been MaxAge'd recently and not flooded out yet), meaning other routers will use stale link state.

Reply:
As a sidenote, it's possible that this change ended up hiding (or rather, delaying) the real problem, which takes the form of not refreshing the LSA correctly.  When we still have the MaxAge LSA, I think (haven't checked) we go through the "refresh" code path, which was broken. Removing the LSA and installing a new one puts us on the "install" code path, which worked... and then broke 30 minutes later, on refresh.

The only things to ever touch MaxAge LSAs should be the age-out timer and the LSDB code when it installs a new LSA on top.  Bypassing it like it was here (no, this doesn't count as the install case) is a recipe for breaking the entire routing domain.

Reply:
@eqvinox 
The reason ospf_default_originate_lsa_update() is introduced to address an issue where
 `default-information originate` config added and removed and readded in short duration. Upon remove command LSA is marked for MAX_AGE and waiting for max_age timer to kick in. Since user readded the configuration, the fix flushes the maxaged lsa and originates new LSA.  The maxage lsa refresh waits for timer to expire than traverse in chunk to flush them out. There isn't a mechanism to check if this needs refresh rather flush. I personally don't think it should have one during maxage flush.
When a user preempt , the LSAS should be flushed immediately from DB send send LSA  update to peer.

Reply:
@chiragshah6 originating the new LSA automatically replaces the maxage LSA. `ospf_lsa_install()` handles this as it should.

Please recheck your analysis of the problem, or maybe run this PR and see that it works correctly.

The previous/current behaviour in master is outright harmful to the network.